updated mobile navigation plugin js to account for recent issues with…

mbfitz opened the pull request:
… scrolling and missing back buttons. also added documentation in sassdoc


@marcelmoreau @khamer @TristanMNorton @aaronmclayton @RustlerCJ 

let me know if you have any feedback or want to see other changes.

TommyChanthaboune committed:
fixed typo in jQuery plugin

Commit:
fixed responsive

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-accordion

* gh-pages:
  Add button and link mixins and styling
  Make search an input[type=search]
  Add button mixin
  Add ::selection
  Add ::selection and ::-moz-selection

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
Merge branch 'gh-pages' into component-dropdown-simple

* gh-pages: (27 commits)
  Add button and link mixins and styling
  Optimized sticky header animation (#61)
  Fix whitespace issues
  Fix sticky header on landing page
  Fixed docker compose command
  Fix indentation in composer.json file
  Add comments to gulpfile
  modified autoprefixer versions
  Fixed typo
  Make search an input[type=search]
  remove hardcoded bg color
  set gutter to margin-left only
  Added dev workflows. gulp tasks, editorconfig, yarn lockfile, etc.
  Add button mixin
  Add ::selection
  missing parens
  wrapped document ready around inline scripts
  adding inline script tags
  Add ::selection and ::-moz-selection
  update css/styles.css
  ...

# Conflicts:
#	.gitignore
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map

Commit:
fixing emmet shorthand; vertically centering actions' elements

Commit:
Merge branch 'gh-pages' into component-login

* gh-pages:
  Add button and link mixins and styling
  Make search an input[type=search]
  Add button mixin
  Add ::selection
  Add ::selection and ::-moz-selection
  update css/styles.css
  remove appearance from buttons
  Add other html5 input types
  Move a.button near button
  Update reset to current elements
  remove @grouping from .container
  Make all hex codes uppercase
  Add grays

# Conflicts:
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #60 from imarc/component-login

adding login component

TommyChanthaboune committed:
update pre, code

Commit:
adding dropdowns to main view

Commit:
removing search form from header;

Commit:
Merge branch 'gh-pages' into component-dropdown-simple

* gh-pages:
  fixing emmet shorthand; vertically centering actions' elements
  fixed responsive
  updated based on Marcel's feedback
  added link to login component
  adding login component

# Conflicts:
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #57 from imarc/component-dropdown-simple

adding simple dropdown menu option

Commit:
fixed path to js

Commit:
ensuring that stickyHeader plugin is called

Commit:
fixed bug in sticky header plugin resize function

Commit:
added node_modules to exclude

Commit:
Merge branch 'gh-pages' into search-listing

* gh-pages: (58 commits)
  fixed bug in sticky header plugin resize function
  ensuring that stickyHeader plugin is called
  fixed path to js
  removing search form from header;
  adding dropdowns to main view
  fixing emmet shorthand; vertically centering actions' elements
  fixed responsive
  updated based on Marcel's feedback
  added node_modules
  Add button and link mixins and styling
  Optimized sticky header animation (#61)
  added link to login component
  Fix whitespace issues
  Fix sticky header on landing page
  Fixed docker compose command
  Fix indentation in composer.json file
  Add comments to gulpfile
  modified autoprefixer versions
  adding login component
  Fixed typo
  ...

# Conflicts:
#	.gitignore
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

marcelmoreau committed:
Merge pull request #50 from imarc/search-listing

adding search listing component

Commit:
adding pagination component

Commit:
exclude node_modules

Commit:
features() should nowrap; updating fontawesome version

Commit:
adding tiles component

Commit:
nowrap our features

Commit:
cleaning up descriptions

Commit:
Merge branch 'component-tiles' into gh-pages

* component-tiles:
  cleaning up descriptions
  adding tiles component

# Conflicts:
#	css/styles.css.map

Commit:
adding dossier component

Commit:
adding $photoWidth param

Commit:
adding default $gutter value for features mixin

Commit:
adding testimonial placeholder to components>index page

TommyChanthaboune committed:
Update markup, Update Documentation, Update jQuery plugin

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-accordion

* gh-pages: (23 commits)
  adding testimonial placeholder to components>index page
  adding default $gutter value for features mixin
  cleaning up descriptions
  nowrap our features
  adding tiles component
  features() should nowrap; updating fontawesome version
  adding pagination component
  added node_modules to exclude
  fixed bug in sticky header plugin resize function
  ensuring that stickyHeader plugin is called
  fixed path to js
  removing search form from header;
  adding dropdowns to main view
  fixing emmet shorthand; vertically centering actions' elements
  fixed responsive
  updated based on Marcel's feedback
  added node_modules
  added link to login component
  adding login component
  added new styles to dropdown
  ...

# Conflicts:
#	components/index.html
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

TommyChanthaboune committed:
committing styles.css and map file for merge

Commit:
created filters component

TommyChanthaboune committed:
Remove header tags in .content, update emmet, update /components, update font awesome in component

Commit:
starting scroll component

TommyChanthaboune committed:
Update accordions to accordion,
Reduce comment block size,
In _components.scss move accordion mixin to top

Commit:
minor tweak to format style

TommyChanthaboune committed:
Merge pull request #53 from imarc/component-accordion

Component Accordion

Commit:
starting on testimonial component!

Commit:
Merge branch 'component-accordion' into gh-pages

* component-accordion:
  minor tweak to format style

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-mobile-nav

* gh-pages: (60 commits)
  adding testimonial placeholder to components>index page
  adding default $gutter value for features mixin
  cleaning up descriptions
  nowrap our features
  adding tiles component
  features() should nowrap; updating fontawesome version
  adding pagination component
  added node_modules to exclude
  fixed bug in sticky header plugin resize function
  ensuring that stickyHeader plugin is called
  fixed path to js
  removing search form from header;
  adding dropdowns to main view
  fixing emmet shorthand; vertically centering actions' elements
  fixed responsive
  updated based on Marcel's feedback
  added node_modules
  Add button and link mixins and styling
  Optimized sticky header animation (#61)
  added link to login component
  ...

# Conflicts:
#	components/index.html
#	components/sticky-header.html
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss
#	index.html
#	js/site.js
#	js/sticky-header.plugin.js
#	js/stickyHeader.jquery.js
#	js/stickyHeader.plugin.js

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-mobile-nav

* gh-pages:
  minor tweak to format style
  Update accordions to accordion, Reduce comment block size, In _components.scss move accordion mixin to top
  Remove header tags in .content, update emmet, update /components, update font awesome in component
  Update markup, Update Documentation, Update jQuery plugin
  exclude node_modules
  update pre, code
  fixed typo in jQuery plugin
  Revert "fixed typo"
  fixed typo
  - please update emmet in HTML file and SCSS with whatever you go with. Right now SCSS shows old. Also, it does not output an h3 - comb _components.scss file for alphabetical - first :after pseudo is missing double colon - rename your plugin file name to match the existing naming convention of thing.plugin.js
  center icons in accordion button
  update code example
  Fix indentation, change accordion class names, use font awesome mixin, update documentation to use th, reworked plugin to not show an icon if Js is unavailable.
  Change html title
  Accordion jQuery Plugin, Accordion Scss Mixin, Added styles to <code>,

# Conflicts:
#	css/_components.scss
#	css/styles.css.map

Commit:
clarifying mega dropdown  menu documentation

khamer committed:
Implement sticky element plugin and component example

khamer committed:
Add link and touch up docs

Commit:
starting work

Commit:
ensure any last-child has no bottom margin in tiles

TommyChanthaboune committed:
resolve conflicts

Commit:
Merge branch 'gh-pages' into component-filters

# Conflicts:
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map

Commit:
removing generated css and cssmap from repo

TommyChanthaboune committed:
fix some styling issues from other merges

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-mobile-nav

* gh-pages:
  removing generated css and cssmap from repo

# Conflicts:
#	css/styles.css
#	css/styles.css.map

Commit:
Merge branch 'gh-pages' into component-dossier

* gh-pages:
  removing generated css and cssmap from repo
  ensure any last-child has no bottom margin in tiles
  clarifying mega dropdown  menu documentation
  minor tweak to format style
  Update accordions to accordion, Reduce comment block size, In _components.scss move accordion mixin to top
  Remove header tags in .content, update emmet, update /components, update font awesome in component
  Update markup, Update Documentation, Update jQuery plugin
  adding testimonial placeholder to components>index page
  adding default $gutter value for features mixin
  exclude node_modules
  update pre, code
  fixed typo in jQuery plugin
  Revert "fixed typo"
  fixed typo
  - please update emmet in HTML file and SCSS with whatever you go with. Right now SCSS shows old. Also, it does not output an h3 - comb _components.scss file for alphabetical - first :after pseudo is missing double colon - rename your plugin file name to match the existing naming convention of thing.plugin.js
  center icons in accordion button
  update code example
  Fix indentation, change accordion class names, use font awesome mixin, update documentation to use th, reworked plugin to not show an icon if Js is unavailable.
  Change html title
  Accordion jQuery Plugin, Accordion Scss Mixin, Added styles to <code>,

# Conflicts:
#	components/index.html
#	css/styles.css
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #65 from imarc/component-dossier

Component dossier

khamer committed:
remove unused pagination mixin

khamer committed:
Merge branch 'gh-pages' into nav-variants

khamer committed:
Remove errant extra space

Commit:
deleting generated css files; fixing links on component index;

Commit:
updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html

Commit:
made some comment edits and wrapped component html page in a container

TommyChanthaboune committed:
Add overlay style and behavior, remove 4th level,

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-mobile-nav

* gh-pages:
  deleting generated css files; fixing links on component index;
  adding $photoWidth param
  adding dossier component

khamer committed:
implement a nav.breadcrumbs, nav.progress, plain nav

khamer committed:
remove examples not part of these components

khamer committed:
Clean up and documentation

Commit:
listing component clean up

Commit:
adding stylesheet back to master

Commit:
Merge branch 'gh-pages' into component-filters

* gh-pages:
  adding stylesheet back to master
  deleting generated css files; fixing links on component index;
  removing generated css and cssmap from repo
  adding $photoWidth param
  adding dossier component

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
removing unnecessary componenta

marcelmoreau committed:
Merge pull request #66 from imarc/component-filters

Filters component

khamer committed:
Set indent size to 4 spaces

Commit:
finished testimonial component and added option with asset

Commit:
committing styles after pull

Commit:
Merge branch 'gh-pages' into component-scroll-to-section

* gh-pages: (22 commits)
  adding stylesheet back to master
  deleting generated css files; fixing links on component index;
  removing generated css and cssmap from repo
  ensure any last-child has no bottom margin in tiles
  clarifying mega dropdown  menu documentation
  minor tweak to format style
  Update accordions to accordion, Reduce comment block size, In _components.scss move accordion mixin to top
  Remove header tags in .content, update emmet, update /components, update font awesome in component
  Update markup, Update Documentation, Update jQuery plugin
  adding $photoWidth param
  adding dossier component
  exclude node_modules
  update pre, code
  fixed typo in jQuery plugin
  Revert "fixed typo"
  fixed typo
  - please update emmet in HTML file and SCSS with whatever you go with. Right now SCSS shows old. Also, it does not output an h3 - comb _components.scss file for alphabetical - first :after pseudo is missing double colon - rename your plugin file name to match the existing naming convention of thing.plugin.js
  center icons in accordion button
  update code example
  Fix indentation, change accordion class names, use font awesome mixin, update documentation to use th, reworked plugin to not show an icon if Js is unavailable.
  ...

# Conflicts:
#	css/styles.css.map

Commit:
Merge branch 'gh-pages' into component-scroll-to-section

* gh-pages:
  committing styles after pull
  Set indent size to 4 spaces
  removing unnecessary componenta
  listing component clean up
  updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html
  created filters component

# Conflicts:
#	css/styles.css
#	css/styles.css.map

Commit:
linked to component

Commit:
fixing paths on dossier images

Commit:
working on marcel's feedback

TommyChanthaboune committed:
improved overlay, add documentation

TommyChanthaboune committed:
Merge branch 'gh-pages' into component-mobile-nav

* gh-pages:
  fixing paths on dossier images
  committing styles after pull
  Set indent size to 4 spaces
  removing unnecessary componenta
  adding stylesheet back to master
  listing component clean up
  updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html
  created filters component

# Conflicts:
#	components/index.html
#	css/styles.css
#	css/styles.css.map

TommyChanthaboune committed:
update css and map files

marcelmoreau committed:
Merge pull request #54 from imarc/component-mobile-nav

Component mobile nav

Commit:
removed mobile nav instance from being called on a generic .filters element; minor wording edits

Commit:
modified emmet and placeholder image text

Commit:
changing default nav anchor color

Commit:
adding plugin file

jeffturcotte committed:
Removed console.log, Fixed iOS "end-of-scroll" stick

Commit:
incorporated all of marcel's feedback EXCEPT the font-size, i think it benefits being smaller on mobile because otherwise it's huuuggeee and takes up too much space

Commit:
Merge branch 'gh-pages' into sticky-element

* gh-pages: (41 commits)
  removed mobile nav instance from being called on a generic .filters element; minor wording edits
  update css and map files
  improved overlay, add documentation
  fixing paths on dossier images
  committing styles after pull
  Set indent size to 4 spaces
  removing unnecessary componenta
  adding stylesheet back to master
  listing component clean up
  Add overlay style and behavior, remove 4th level,
  updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html
  deleting generated css files; fixing links on component index;
  fix some styling issues from other merges
  removing generated css and cssmap from repo
  ensure any last-child has no bottom margin in tiles
  clarifying mega dropdown  menu documentation
  minor tweak to format style
  Update accordions to accordion, Reduce comment block size, In _components.scss move accordion mixin to top
  Remove header tags in .content, update emmet, update /components, update font awesome in component
  created filters component
  ...

# Conflicts:
#	components/index.html
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map

Commit:
Merge branch 'sticky-element' into gh-pages

* sticky-element:
  Removed console.log, Fixed iOS "end-of-scroll" stick
  made some comment edits and wrapped component html page in a container
  Add link and touch up docs
  Implement sticky element plugin and component example

Commit:
fixed chrome flex box bug

Commit:
deleted erroneous css file - gulp bug

Commit:
refactoring and cleanup

Commit:
Merge branch 'gh-pages' into component-testimonial

* gh-pages: (45 commits)
  Removed console.log, Fixed iOS "end-of-scroll" stick
  removed mobile nav instance from being called on a generic .filters element; minor wording edits
  update css and map files
  improved overlay, add documentation
  fixing paths on dossier images
  committing styles after pull
  Set indent size to 4 spaces
  removing unnecessary componenta
  adding stylesheet back to master
  listing component clean up
  Add overlay style and behavior, remove 4th level,
  made some comment edits and wrapped component html page in a container
  updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html
  deleting generated css files; fixing links on component index;
  fix some styling issues from other merges
  removing generated css and cssmap from repo
  ensure any last-child has no bottom margin in tiles
  Add link and touch up docs
  Implement sticky element plugin and component example
  clarifying mega dropdown  menu documentation
  ...

# Conflicts:
#	components/index.html
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #68 from imarc/component-testimonial

Component testimonial

Commit:
Merge branch 'gh-pages' into nav-variants

* gh-pages: (35 commits)
  refactoring and cleanup
  deleted erroneous css file - gulp bug
  fixed chrome flex box bug
  incorporated all of marcel's feedback EXCEPT the font-size, i think it benefits being smaller on mobile because otherwise it's huuuggeee and takes up too much space
  Removed console.log, Fixed iOS "end-of-scroll" stick
  modified emmet and placeholder image text
  removed mobile nav instance from being called on a generic .filters element; minor wording edits
  update css and map files
  improved overlay, add documentation
  working on marcel's feedback
  fixing paths on dossier images
  committing styles after pull
  finished testimonial component and added option with asset
  Set indent size to 4 spaces
  removing unnecessary componenta
  adding stylesheet back to master
  listing component clean up
  Add overlay style and behavior, remove 4th level,
  made some comment edits and wrapped component html page in a container
  updated search listing component to include search filters as well. now on combo component on search and filters! removed filters.html and renamed search-listing.html to search-filters.html
  ...

# Conflicts:
#	css/_components.scss
#	css/styles.css
#	css/styles.css.map

Commit:
clean up

TommyChanthaboune committed:
Mobile Menu overflow-y

TommyChanthaboune committed:
Mobile Menu overflow-y #69

Commit:
mobile menu cosmetic edits

Commit:
attempting to resolve scroll issue w/ overlay

TommyChanthaboune committed:
fix body scroll issue, exposed toggleAll method

Commit:
adding smooth-scroll script documentation

Commit:
removing native webkit styling to three form elements

Commit:
Merge branch 'gh-pages' into component-mobile-nav-two

* gh-pages:
  removing native webkit styling to three form elements

# Conflicts:
#	css/styles.css.map

Commit:
mobile nav's search form styling; general stylesheet formatting;

Commit:
refactor of css rules;

Commit:
Merge branch 'gh-pages' into real-world-header

* gh-pages:
  removing native webkit styling to three form elements
  mobile menu cosmetic edits
  Mobile Menu overflow-y #69
  Mobile Menu overflow-y

TommyChanthaboune committed:
move the mixin calls to the Components section of the SASS file with everything else

Commit:
added flexbox to header; more nav items; added search pattern

Commit:
renaming entire site wrapper

Commit:
moved mobilenavigation mixin call to components area

Commit:
adding flexbox to page structure

Commit:
Merge branch 'gh-pages' into component-mobile-nav-two

* gh-pages:
  adding flexbox to page structure

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
Merge branch 'component-mobile-nav-two' into gh-pages

* component-mobile-nav-two:
  moved mobilenavigation mixin call to components area
  renaming entire site wrapper
  move the mixin calls to the Components section of the SASS file with everything else
  refactor of css rules;
  mobile nav's search form styling; general stylesheet formatting;
  fix body scroll issue, exposed toggleAll method
  attempting to resolve scroll issue w/ overlay

Commit:
Merge branch 'gh-pages' into real-world-header

* gh-pages:
  adding flexbox to page structure
  moved mobilenavigation mixin call to components area
  renaming entire site wrapper
  move the mixin calls to the Components section of the SASS file with everything else
  refactor of css rules;
  mobile nav's search form styling; general stylesheet formatting;
  fix body scroll issue, exposed toggleAll method
  attempting to resolve scroll issue w/ overlay

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss
#	index.html

Commit:
Merge branch 'real-world-header' into gh-pages

* real-world-header:
  added flexbox to header; more nav items; added search pattern

Commit:
generated styles

Commit:
Merge branch 'gh-pages' into component-scroll-to-section

* gh-pages: (39 commits)
  generated styles
  adding flexbox to page structure
  moved mobilenavigation mixin call to components area
  renaming entire site wrapper
  added flexbox to header; more nav items; added search pattern
  move the mixin calls to the Components section of the SASS file with everything else
  refactor of css rules;
  mobile nav's search form styling; general stylesheet formatting;
  removing native webkit styling to three form elements
  fix body scroll issue, exposed toggleAll method
  attempting to resolve scroll issue w/ overlay
  mobile menu cosmetic edits
  Mobile Menu overflow-y #69
  Mobile Menu overflow-y
  refactoring and cleanup
  deleted erroneous css file - gulp bug
  fixed chrome flex box bug
  incorporated all of marcel's feedback EXCEPT the font-size, i think it benefits being smaller on mobile because otherwise it's huuuggeee and takes up too much space
  Removed console.log, Fixed iOS "end-of-scroll" stick
  modified emmet and placeholder image text
  ...

# Conflicts:
#	components/tabs.html
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

TommyChanthaboune committed:
refactoring simple stuff like adding ‘use strict’ and adding comments

Commit:
further documentation; moving script file into a lib directory

khamer committed:
Convert to use sassdoc compatible comment format

Commit:
updating html headers on components with new real-world header

Commit:
Merge branch 'gh-pages' into javascript-refactor

* gh-pages:
  updating html headers on components with new real-world header

# Conflicts:
#	css/styles.css.map

Commit:
spacing changes

marcelmoreau committed:
Merge pull request #77 from imarc/javascript-refactor

Javascript refactor

Commit:
fixing image paths

Commit:
updating jquery; continuity in footer markup across views; moved components section in styles.scss higher top top; refactored .social and footer.primary styles

khamer committed:
only reference that when referencing this won't work

khamer committed:
indicate variables containing jQuery objects with $

khamer committed:
simplify this.update and this.fix logic

khamer committed:
make $until clearly a jQuery object

khamer committed:
better docs in the example

Commit:
Merge branch 'gh-pages' into component-scroll-to-section

* gh-pages:
  updating jquery; continuity in footer markup across views; moved components section in styles.scss higher top top; refactored .social and footer.primary styles
  fixing image paths
  spacing changes
  updating html headers on components with new real-world header
  refactoring simple stuff like adding ‘use strict’ and adding comments

# Conflicts:
#	components/tabs.html
#	css/styles.css
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #70 from imarc/component-scroll-to-section

Component scroll to section

Commit:
updating browser version support

Commit:
Merge branch 'gh-pages' into nav-variants

* gh-pages: (27 commits)
  updating browser version support
  updating jquery; continuity in footer markup across views; moved components section in styles.scss higher top top; refactored .social and footer.primary styles
  fixing image paths
  spacing changes
  updating html headers on components with new real-world header
  further documentation; moving script file into a lib directory
  refactoring simple stuff like adding ‘use strict’ and adding comments
  generated styles
  adding flexbox to page structure
  moved mobilenavigation mixin call to components area
  renaming entire site wrapper
  added flexbox to header; more nav items; added search pattern
  move the mixin calls to the Components section of the SASS file with everything else
  refactor of css rules;
  mobile nav's search form styling; general stylesheet formatting;
  removing native webkit styling to three form elements
  adding smooth-scroll script documentation
  fix body scroll issue, exposed toggleAll method
  attempting to resolve scroll issue w/ overlay
  mobile menu cosmetic edits
  ...

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
new progress breadcrumbs

marcelmoreau committed:
Merge pull request #80 from imarc/sticky-element-cleanup

Sticky element cleanup

Commit:
fixes to dropdown navs

Commit:
Merge branch 'gh-pages' into nav-variants

* gh-pages:
  better docs in the example
  make $until clearly a jQuery object
  simplify this.update and this.fix logic
  indicate variables containing jQuery objects with $
  only reference that when referencing this won't work

Commit:
breaking out breadcrumbs and progress breadcrumbs

marcelmoreau committed:
Merge pull request #51 from imarc/nav-variants

Nav variants

Commit:
fixes to sticky element and footer fixes

khamer committed:
Setting rev to 3.0

Commit:
new hero images

Commit:
updated tiles’ negative wrapping margin to match .container

khamer committed:
fix package.json

TommyChanthaboune committed:
fix  bug w/ sticky header & mobile menu when used together

TommyChanthaboune committed:
accordion should be structured differently #81

Commit:
better readme content relevant to existing boilerplate

Commit:
Merge branch 'gh-pages' into bug-fixes-02-24-17

* gh-pages:
  better readme content relevant to existing boilerplate

Commit:
Merge branch 'bug-fixes-02-24-17' into gh-pages

* bug-fixes-02-24-17:
  accordion should be structured differently #81
  fix  bug w/ sticky header & mobile menu when used together

khamer committed:
first pass working sassdocs with custom theme

khamer committed:
Merge branch 'gh-pages' into sassdoc

khamer committed:
Addding in demo links

khamer committed:
clean up of variable docs

khamer committed:
Update READMEs

khamer committed:
changing to not depend on Makefile

Commit:
updating components usage docs

Commit:
adding tab styles back

khamer committed:
Add untilVisible option to stickyElement.plugin.js

khamer committed:
typo and remove a console.log

Commit:
adding 0-width gutter comment

khamer committed:
Move imarc-theme to imarc/boilerplate-sassdoc-theme repo

khamer committed:
Updating boilerplate to reference sassdoc-theme-boilerplate

khamer committed:
Merge branch 'gh-pages' into sassdoc

khamer committed:
Merge pull request #84 from imarc/sassdoc

SassDoc Documentation

khamer committed:
Removing old, irrelevant instructions from README.md

Commit:
making tiles' container have calculated negative margin

khamer committed:
up to 3.1

khamer committed:
Update _components.scss to sassdoc comment format errantly removed

Commit:
removing aria-labeling; adding proper doc content for compontents/sassdoc

Commit:
adding more documentation to components

Commit:
IE11 bugs; correcting some emmet and improving documentation for components

Commit:
ie11 flex padding fix on tiles

Commit:
filling in more emmet shorthand

Commit:
improving read me for components

khamer committed:
allow boilerplate to be published to npmjs

khamer committed:
updating name so it can be published to npmjs

khamer committed:
Switch docs to components/; switch demos to demo/;

cherrykoda committed:
Updated web_author tag to remove locations

Commit:
removed mobile navigation call from plugin.js file and moved it into index and mobile navigation demo html files

cherrykoda committed:
Merge pull request #93 from imarc/meta_update

Updated web_author tag to remove locations

marcelmoreau committed:
Merge pull request #94 from imarc/mobile-nav-js-fix

removed mobile navigation call from plugin.js file

Commit:
increasing version

Commit:
ran ‘nom run docs’ and hoping this pushes components changes

Commit:
no longer need -moz- on placeholders

Commit:
tweaking fontawesome to allow for always-centered backgrounds

Commit:
editing command that generates sassdoc to the correct command

Commit:
adding aria-hidden attr to dropdown arrow that did not have it. Adding aria-labels to both ‘level’ buttons; adding a unit to flex-basis in a flex shorthand for IE11

Commit:
aria-labels unnecessary here

Commit:
Merge branch 'gh-pages' into social-icons-centering

* gh-pages:
  aria-labels unnecessary here
  adding aria-hidden attr to dropdown arrow that did not have it. Adding aria-labels to both ‘level’ buttons; adding a unit to flex-basis in a flex shorthand for IE11
  editing command that generates sassdoc to the correct command

# Conflicts:
#	css/styles.css.map

marcelmoreau committed:
Merge pull request #98 from imarc/social-icons-centering

tweaking fontawesome to allow for always-centered backgrounds

Commit:
matching mobile and desktop nav markup; carrying changes over to other views; modified selector for mobilenav plugin to look for;

Commit:
updating mega dropdown’s emmet and markup for sassdocs

marcelmoreau committed:
Merge pull request #100 from imarc/mobile-nav-markup-unity

matching mobile and desktop nav markup;

Commit:
cosmetic changes to Components area

Commit:
darkened default border, added height to select, added padding to select[multiple]

jeffturcotte committed:
Removed block style from .help class and updated example

marcelmoreau committed:
Merge pull request #102 from imarc/help-fix

Removed block style from .help class and updated example

marcelmoreau committed:
Merge pull request #101 from imarc/border-select-element

darkened default border, added height to select, added padding to sel…

keyesatimarc committed:
minor tweaks to boilerplate per my thoughts and preferences.

Commit:
deleting searchDrawer js plugin and adding it into our sticky header plugin; ensuring search drawer is visible when you are down a ways on a page

keyesatimarc committed:
Adding my responsive tester and reconfiguring break points to allow for a bit more flexibility.

Commit:
Task #91:  #91: fix the site messaging icons

Commit:
gave prism media queries size-names instead of color and reduced amount of queries; removed $rem var; remove block display on images

khamer committed:
add class while element is stuck

Commit:
Merge branch 'add-class-to-sticky-elements' into gh-pages

* add-class-to-sticky-elements:
  add class while element is stuck

Commit:
Merge branch 'gh-pages' into sticky-header-fixes-2017-10

* gh-pages:
  add class while element is stuck

marcelmoreau committed:
Merge pull request #103 from imarc/sticky-header-fixes-2017-10

deleting searchDrawer js plugin and adding it into our sticky header …

marcelmoreau committed:
Merge pull request #103 from imarc/sticky-header-fixes-2017-10

deleting searchDrawer js plugin and adding it into our sticky header …

Commit:
Merge branch 'gh-pages' into James-Revisions

Commit:
Merge branch 'gh-pages' into James-Revisions

Commit:
Task #91:  #91: fix the site messaging icons

Commit:
Task #91:  #91: fix the site messaging icons

marcelmoreau committed:
Merge pull request #105 from imarc/James-Revisions

 Task #91:  #91: fix the site messaging icons

marcelmoreau committed:
Merge pull request #105 from imarc/James-Revisions

 Task #91:  #91: fix the site messaging icons

Commit:
minor naming changes; removing conditional from transition mixin

Commit:
Merge branch 'gh-pages' into keyes-suggestions-marcel

* gh-pages:
  Task #91:  #91: fix the site messaging icons
  add class while element is stuck
  Task #91:  #91: fix the site messaging icons
  deleting searchDrawer js plugin and adding it into our sticky header plugin; ensuring search drawer is visible when you are down a ways on a page

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

Commit:
Merge branch 'gh-pages' into keyes-suggestions-marcel

* gh-pages:
  Task #91:  #91: fix the site messaging icons
  add class while element is stuck
  Task #91:  #91: fix the site messaging icons
  deleting searchDrawer js plugin and adding it into our sticky header plugin; ensuring search drawer is visible when you are down a ways on a page

# Conflicts:
#	css/styles.css
#	css/styles.css.map
#	css/styles.scss

marcelmoreau committed:
Merge pull request #107 from imarc/keyes-suggestions-marcel

Keyes suggestions marcel

marcelmoreau committed:
Merge pull request #107 from imarc/keyes-suggestions-marcel

Keyes suggestions marcel

wickedgenius committed:
Normalize as a reset

wickedgenius committed:
Normalize as a reset

aaronmclayton committed:
Remove commented code in Base area

aaronmclayton committed:
Remove commented code in Base area

wickedgenius committed:
Add second colon for a pseudo elements

wickedgenius committed:
Add second colon for a pseudo elements

wickedgenius committed:
Import normalize-scss npm module. Re-build and commit css files, map and package.json

wickedgenius committed:
Import normalize-scss npm module. Re-build and commit css files, map and package.json

wickedgenius committed:
Change font stack back to original

wickedgenius committed:
Change font stack back to original

wickedgenius committed:
Fieldset guideline corrections

wickedgenius committed:
Fieldset guideline corrections

marcelmoreau committed:
Merge pull request #108 from imarc/NormalizeCSS

Normalize as a reset

marcelmoreau committed:
Merge pull request #108 from imarc/NormalizeCSS

Normalize as a reset

Commit:
wip

Commit:
fixes to header primary’s search drawer

Commit:
wip

Commit:
ensuring components link to correct js. changing icons to font awesome v5; larval mix work

Commit:
cleaning up component files. ensuring component/docs can be run

Commit:
cleaning up components

Commit:
Merge branch 'gh-pages' into laravel-mix-and-fontawesome-v5

* gh-pages:
  fixes to header primary’s search drawer
  Fieldset guideline corrections
  Change font stack back to original
  Import normalize-scss npm module. Re-build and commit css files, map and package.json
  Add second colon for a pseudo elements
  Remove commented code in Base area
  Normalize as a reset
  Task #91:  #91: fix the site messaging icons

# Conflicts:
#	css/app.css
#	css/styles.css.map
#	demo/dropdown-simple.html
#	index.html
#	package-lock.json
#	package.json
#	resources/assets/sass/app.scss

Commit:
updated readme; modified fontawsome pseudo element mixin for v5; fixed site messaging iconography;

Commit:
condensing js and sass methods in webpack.mix.js like order my oil’s; added some obscure prop/value to webpack config to try to get sourcemaps to work

Commit:
sourcemap work

Commit:
use fontawesome free tier by default

marcelmoreau committed:
Merge pull request #109 from imarc/laravel-mix-and-fontawesome-v5

Laravel mix and fontawesome v5

Commit:
updated mobile navigation plugin js to account for recent issues with scrolling and missing back buttons. also added documentation in sassdoc

khamer commented:
So, two things –

- I'm not sure why we're changing from .wrapper to .shell; it's fine if there's a reason, but I don't see any sass styles that would target a .wrapper or .shell in the header or .torso? I see .wrapper in the footer and testimonial sass...
- `'>' + subMenuClass + '> div'` is more fragile than than `subMenu+' ul'`; I think in an older version, I pushed to not use any direct child selectors in plugin. Maybe something like `subMenu + ' ' + subMenuDiv`, setting up another class and variable? I'd have to dig further into to see if that makes sense.

mbfitz commented:
@khamer - I changed it to .shell because that's what all of our HTML is using, I believe that got changed from .wrapper because .wrapper is more common and may cause conficts, but the documentation hadn't been updated, even though the markup was. 

re: the sub menu selector - I think it's safe to say we use class .menu within the `subMenuClass`, so maybe we target that instead? I just used the code I've used to fix it a few times, but I've also updated the markup to include the .menu class, so that should work ok. Thoughts?

marcelmoreau commented:
`.shell` is used on purpose to be unique. at the time, the discussion point was that `.wrapper` was/is often used (as a flex container) as a generic class in many places in a site. boilerplate does not exemplify a site that requires a lot of layout. we wanted to avoid conflict by having the site wrapper be something unique.

yes @mbfitz please use that `.menu` class test. I agree the existing query is super specific

Can we update the fontawesome markup in the plugin file? There should be a font weight prefix on the `.fa` class such as `.far` or `.fas`. We also do not need the aria attribute any more. 

Please make sure this works on the boilerplate home page in addition to the demo page
